src/shogun/mathematics/linalg/LinalgBackendEigen.h

```diff
@@ -111,6 +133,18 @@ class LinalgBackendEigen : public LinalgBackendBase
 		return (typename SGVector<T>::EigenVectorXtMap(a)).dot(typename SGVector<T>::EigenVectorXtMap(b));
 	}
 
+	/** Eigen3 vector scale method: B = alpha * A */
+	template <typename T>
+	SGVector<T> scale_impl(const SGVector<T>& a, T alpha) const
+	{
+		SGVector<T> b(a.vlen);
```

Comment:
I vote that for now, we *always* do these type of operations in place.
And if a developer doesnt want that, he can copy the vector first